components/kyma-environment-broker/cmd/broker/main.go

```diff
@@ -687,13 +694,60 @@ func NewProvisioningProcessingQueue(ctx context.Context, provisionManager *provi
 }
 
 func NewUpdateProcessingQueue(ctx context.Context, manager *update.Manager, workersAmount int, db storage.BrokerStorage, inputFactory input.CreatorForPlan,
-	provisionerClient provisioner.Client, publisher event.Publisher, logs logrus.FieldLogger) *process.Queue {
+	provisionerClient provisioner.Client, publisher event.Publisher, runtimeVerConfigurator *runtimeversion.RuntimeVersionConfigurator, runtimeStatesDb storage.RuntimeStates,
+	runtimeProvider input.ComponentListProvider, reconcilerClient reconciler.Client, logs logrus.FieldLogger) *process.Queue {
 
-	manager.DefineStages([]string{"cluster", "check"})
-	manager.AddStep("cluster", update.NewInitialisationStep(db.Instances(), db.Operations(), inputFactory))
-	manager.AddStep("cluster", update.NewUpgradeShootStep(db.Operations(), db.RuntimeStates(), provisionerClient))
-	manager.AddStep("check", update.NewCheckStep(db.Operations(), provisionerClient, 40*time.Minute))
+	manager.DefineStages([]string{"cluster", "runtime", "check"})
+	updateSteps := []struct {
+		stage     string
+		step      update.Step
+		condition update.StepCondition
+	}{
+		{
+			stage: "cluster",
+			step:  update.NewInitialisationStep(db.Instances(), db.Operations(), inputFactory, runtimeVerConfigurator, runtimeStatesDb),
+		},
+		{
+			stage: "cluster",
+			step:  update.NewUpgradeShootStep(db.Operations(), db.RuntimeStates(), provisionerClient),
+		},
+		{
+			stage:     "runtime",
+			step:      update.NewBTPOperatorOverridesStep(runtimeProvider),
+			condition: update.ForBTPOperatorCredentialsProvided,
+		},
+		{
+			stage:     "runtime",
+			step:      update.NewSCMigrationStep(runtimeProvider),
+			condition: update.ForMigration,
+		},
+		{
+			stage:     "runtime",
+			step:      update.NewApplyReconcilerConfigurationStep(db.Operations(), db.RuntimeStates(), reconcilerClient),
+			condition: update.ForKyma2,
+		},
+		{
+			stage:     "runtime",
+			step:      update.NewCheckSCMigrationDone(reconcilerClient),
+			condition: update.ForKyma2,
+		},
+		{
+			stage:     "runtime",
+			step:      update.NewApplyReconcilerConfigurationStep(db.Operations(), db.RuntimeStates(), reconcilerClient),
+			condition: update.ForKyma2,
+		},
+		{
+			stage: "check",
+			step:  update.NewCheckStep(db.Operations(), provisionerClient, 40*time.Minute),
+		},
```

Comment:
the new steps for update perform
- fetch lastest `runtime_state`
- if `btp-operator` credentials are provided instal `btp-operator` component
- if migration flag is provided, install `sc-migration` component
- call reconciler with modified payload (list of components and overrides contain `btp-operator` and/or `sc-migration`)
- keep checking the reconciler whether the migration is done, `sc-migration` should not claim `reconciled` unless the migration has not finished
- remove SVCAT related components
- call reconciler with modified payload (removed SVCAT components)